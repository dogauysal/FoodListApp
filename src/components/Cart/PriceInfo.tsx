import React, { useMemo } from "react";
import { StyleSheet, Text, View } from "react-native"

interface IProps {
    gross: number;
    discount: number;
}

const PriceInfo: React.FC<IProps> = ({
    gross,
    discount
}) => {

    const total = useMemo(() => { return gross - discount }, [gross, discount])

    return (
        <View style={styles.container}>
            <View style={styles.row}>
                <Text style={styles.price}>Fiyat</Text>
                <Text style={styles.price}>{gross} TL</Text>
                {discount > 0 && <View style={styles.crossLine} />}
            </View>
            {discount > 0 && (
                <View style={styles.row}>
                    <Text style={styles.price}>Ä°ndirim</Text>
                    <Text style={styles.price}>{discount} TL</Text>
                </View>
            )}

            <View style={styles.divider}></View>
            <View style={styles.row}>
                <Text style={styles.price}>Toplam</Text>
                <Text style={styles.price}> {total} TL</Text>
            </View>
        </View>
    )
}

const styles = StyleSheet.create({
    container: {
        padding: 15,
        // width: "50%"
    },
    row: {
        marginVertical: 10,
        paddingHorizontal: 2,
        flexDirection: "row",
        justifyContent: "space-between"
    },
    price: {
        fontSize: 24
    },
    divider: {
        borderWidth: 1,
        marginHorizontal: -5
    },
    crossLine: {
        width: 48,
        height: 4,
        backgroundColor: 'black',
        transform: [{ rotate: '-45deg' }],
        position: "absolute",
        right: 24,
        bottom: 16
    },
})

export default PriceInfo;