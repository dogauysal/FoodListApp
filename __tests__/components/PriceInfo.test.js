const { describe, test, expect } = require("@jest/globals");
import renderer from 'react-test-renderer';
import PriceInfo from '../../src/components/Cart/PriceInfo';

describe("PriceInfo", () => {
    test("PriceInfo renders correctly", () => {
        const gross = 30
        const discount = 10

        const priceInfo = renderer.create(<PriceInfo gross={gross} discount={discount} />).toJSON()

        expect(priceInfo).toMatchSnapshot()
    })
})