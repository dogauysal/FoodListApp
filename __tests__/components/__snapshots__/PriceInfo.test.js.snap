// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PriceInfo PriceInfo renders correctly 1`] = `
<View
  style={
    {
      "padding": 15,
    }
  }
>
  <View
    style={
      {
        "flexDirection": "row",
        "justifyContent": "space-between",
        "marginVertical": 10,
        "paddingHorizontal": 2,
      }
    }
  >
    <Text
      style={
        {
          "fontSize": 24,
        }
      }
    >
      Fiyat
    </Text>
    <Text
      style={
        {
          "fontSize": 24,
        }
      }
    >
      30
       TL
    </Text>
    <View
      style={
        {
          "backgroundColor": "black",
          "bottom": 16,
          "height": 4,
          "position": "absolute",
          "right": 24,
          "transform": [
            {
              "rotate": "-45deg",
            },
          ],
          "width": 48,
        }
      }
    />
  </View>
  <View
    style={
      {
        "flexDirection": "row",
        "justifyContent": "space-between",
        "marginVertical": 10,
        "paddingHorizontal": 2,
      }
    }
  >
    <Text
      style={
        {
          "fontSize": 24,
        }
      }
    >
      Ä°ndirim
    </Text>
    <Text
      style={
        {
          "fontSize": 24,
        }
      }
    >
      10
       TL
    </Text>
  </View>
  <View
    style={
      {
        "borderWidth": 1,
        "marginHorizontal": -5,
      }
    }
  />
  <View
    style={
      {
        "flexDirection": "row",
        "justifyContent": "space-between",
        "marginVertical": 10,
        "paddingHorizontal": 2,
      }
    }
  >
    <Text
      style={
        {
          "fontSize": 24,
        }
      }
    >
      Toplam
    </Text>
    <Text
      style={
        {
          "fontSize": 24,
        }
      }
    >
       
      20
       TL
    </Text>
  </View>
</View>
`;
